### PR TITLE
Don't ask for surface, if footway:surface is already tagged.

### PR DESCRIPTION
**Pull request by Henry00572**
I was asked about the surface of https://www.openstreetmap.org/way/1084496227, which is already tagged with `footway:surface` and `cycleway:surface`. I don't think it needs another `surface` tag.

### CODE REVIEW

**Comment by Robbendebiene** [comment]
Thanks for raising this issue. I agree on this, but I think splitting this question into multiple would be the best way to properly solve this problem:

1. "Wie ist der Boden beschaffen?"
WRITES: `surface`
CONDITION: asked when the way is not segregated or it is unknown whether it is segregated or not (`segregated: [false, "no"]`)
2. "Wie ist der Boden des Fahrradwegs beschaffen?"
WRITES `cycleway:surface`
CONDITION: asked when the way is segregated (`segregated: "yes"`)
3. "Wie ist der Boden des Fußwegs beschaffen?"
WRITES `footway:surface`
CONDITION: asked when the way is segregated (`segregated: "yes"`)

Do you feel comfortable adding this to the `question_catalog.json`? You can also find out more how it works here: https://github.com/OPENER-next/OpenStop/blob/master/docs/QUESTION_CATALOG.md

**Comment by Henry00572** [comment]
Thanks for the help, I think I have created the quests as you said.
The logic works as follows:
1. surface is asked if surface is not set and segregated is ["no", false]. However, for segregated ways where one side is paved and the other one unpaved (I still have to add a picture of this situtation, for example [here](https://maps.duesseldorf.de/?Zoom=10&Center=353068,5666599.4&Marker=&Rotation=0&Background=bg_lubi&Layers=&ExtentsUTM=&PointsUTM=&CirclesUTM=&Lang=de&Nt=false)), segregated=yes is set and the following two quests will ask for the surfaces individually.
2./3, footway:surface and cycleway:surface are asked for highway ["fotway", "cycleway", "path"] with segregated=yes and foot ["yes", "designated"] or bicycle ["yes", "designated"] respectively.

**Comment by Henry00572** [comment]
I have now added an image of the situation where a segregated way has different surfaces. In this case, segregated=yes should be set in the surface quest and the individual surfaces can be inserted by the other new quests.
I have also uploaded the image to [Wikimedia Commons](https://commons.wikimedia.org/wiki/File:Segregated_path_with_different_surfaces_detail.jpg) with CC0 license and I hope it's acceptable.
Thanks again for your help.

**Review by Robbendebiene — COMMENTED** [review]
Thanks for your work. Looks pretty good!

Also great job with adding an image to wiki commons and licensing it with CC0. That is exactly the way we prefer to go. Even though depending on which direction we decide to go we might prefer a different image.

File: assets/datasets/question_catalog.json [hunk 1]
```diff
@@ -1798,6 +1798,13 @@
...
                }
+            },
+            {
+               "name": "Unterschiedlich",
+               "image": "assets/images/questions/segregated_surface_paved_unpaved.jpg",
+               "osm_tags": {
+                  "segregated": "yes"
+               }
             }
          ]
```
[the remark this answers is not in the document]
Comment (Henry00572) [reply assets/datasets/question_catalog.json:1809]: I see the problems with the option `segregated` on platforms and with different surfaces that are not legally segregated. Both could be solved with another quest for the key `segregated`, although with some more logic.

If we add another quest for `segregated`, which is asked on ways and not platforms, then if `segregated=no` is answered, the surface can be answered by the surface quest, and if `segregated=yes` is set then the other new quests for `footway:surface` and `cycleway:surface` apply.

In the case of different surfaces that are not legally segregated, I don't know of any tagging that could describe such a situation and it probably occurs very rarely.

So do you think we should add another quest for `segregated`?

PS: In an accessibility app there should really be a value of `segregated` for describing that a way is segregated using tactile paving, but that's another story.

File: assets/datasets/question_catalog.json [hunk 2]
```diff
@@ -1833,6 +1841,99 @@
...
+            {
+               "name": "Unbefestigt",
+               "image": "assets/images/questions/surface_unpaved.jpg",
+               "osm_tags": {
+                  "footway:surface": "unpaved"
+               }
+            }
+         ]
+      },
+      "conditions": {
```
[the remark this answers is not in the document]
Comment (Henry00572) [reply assets/datasets/question_catalog.json]: Done

File: assets/datasets/question_catalog.json [hunk 3]
```diff
@@ -1833,6 +1841,99 @@
...
```
[the remark this answers is not in the document]
Comment (Henry00572) [reply assets/datasets/question_catalog.json]: Done

File: assets/datasets/question_catalog.json [hunk 4]
```diff
@@ -1833,6 +1841,99 @@
...
+         {
+            "osm_tags": {
+               "highway": ["path", "footway", "cycleway"],
+               "bicycle": ["yes", "designated"],
```
[the remark this answers is not in the document]
Comment (Henry00572) [reply assets/datasets/question_catalog.json]: You're right, `segregated=yes` should be enough, I've removed this condition

File: assets/datasets/question_catalog.json [hunk 5]
```diff
@@ -1833,6 +1841,99 @@
...
+                  "footway:surface": "unpaved"
+               }
+            }
+         ]
+      },
+      "conditions": {
+         {
+            "osm_tags": {
+               "highway": ["path", "footway", "cycleway"],
+               "foot": ["yes", "designated"],
```
[the remark this answers is not in the document]
Comment (Henry00572) [reply assets/datasets/question_catalog.json]: Removed this condition as well.

File: assets/datasets/question_catalog.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Robbendebiene) [reply assets/datasets/question_catalog.json:1809]: > If we add another quest for segregated, which is asked on ways and not platforms, then if segregated=no is answered, the surface can be answered by the surface quest, and if segregated=yes is set then the other new quests for footway:surface and cycleway:surface apply.

> So do you think we should add another quest for segregated?

Yes I think it is the cleanest way even though this means introducing another question.

Perhaps you can make it a `List` answer type with two entries and add the following images:
- https://wiki.openstreetmap.org/wiki/File:Zeichen_241-30_-_getrennter_Rad-_und_Fu%C3%9Fweg,_StVO_1992.svg
- https://wiki.openstreetmap.org/wiki/File:Zeichen_240_-_Gemeinsamer_Fu%C3%9F-_und_Radweg,_StVO_1992.svg

Note though that we unfortunately cannot render svgs so converting them to a png is required. (our preferred resolution is 2000 x 2000px)

I only wonder whether it would make sense to construct a more advanced question that covers more legal cases regarding bicycles and pedestrians. (I'm thinking about some of these scenarios: https://wiki.openstreetmap.org/wiki/DE:Bicycle/Radverkehrsanlagen_kartieren). Or are we losing our selves here? 😅
____________________

> PS: In an accessibility app there should really be a value of segregated for describing that a way is segregated using tactile paving, but that's another story.

Just out of interest, you mean something like this right?

https://www.handi-friendly.com/wp-content/uploads/2021/02/Jirkov-2-scaled.jpg
Or even this https://ecf.com/sites/ecf.com/files/Figure%2038%20v2%20%5BCredit%20Robert%20Weetman%5D.jpg

**Comment by Henry00572** [comment]
> Perhaps you can make it a List answer type with two entries and add the following images

Okay, I think I have done that, I had to resolve some conflicts with #200, but I think that works now.
The conditions are quite strict as the question requires `highway=path/footway/cycleway` + `foot=designated` + `bicycle=designated`. If we wanted to

> construct a more advanced question that covers more legal cases regarding bicycles and pedestrians

this would probably be too strict. For now, the two options `segregated=yes` and `segregated=no` are enough to me.

> Just out of interest, you mean something like this right?

Exactly.

File: assets/datasets/question_catalog.json [hunk 6]
```diff
@@ -1740,6 +1740,50 @@
          }
       ]
    },
+   {
+      "question": {
+         "name": "Trennung",
```
Comment (Robbendebiene) [inline assets/datasets/question_catalog.json:1745]: We prefer: "Fuß-/Radweg-Verlauf"

File: assets/datasets/question_catalog.json [hunk 7]
```diff
@@ -1740,6 +1740,50 @@
...
+         "text": "Wie verlaufen der Fuß- und Radweg zueinander?",
+         "description": "Was für ein Schild schreibt die gemeinsame oder getrennte Nutzung des Weges vor?"
```
Comment (Robbendebiene) [inline assets/datasets/question_catalog.json:1747]: We have a convention that we do not ask questions in the description.
So maybe you can change it to something like this: "Die gemeinsame oder getrennte Nutzung des Weges wird üblicherweise durch Schilder vorgeschrieben."

**Review by Robbendebiene — COMMENTED** [review]
Thanks. I think we are almost there!

The last problem I see is that we now only ever ask any surface questions when `segragated` is set to `yes` or `no`. This itself is not necessarily a problem, but since the conditions for writing segregated ourself are very strict (requiring `"foot": "designated", "bicycle": "designated"`) the surface conditions become likewise as strict.

So we could either loosen the constraints for the segregated question or the surface question.
While removing the `"foot": "designated", "bicycle": "designated",` tags from the segregated question would work, I feel like we rather should tweak the surface questions directly.

For example we could add additional conditions to the questions which ask for the individual surfaces. The following condition would ask the question for `footway:surface` whenever `cycleway:surface` is set:

```json
{
    "osm_tags": {
       "highway": ["path", "footway", "cycleway"],
       "cycleway:surface": true,
       "footway:surface": false,
       "indoor": ["no", false]
    },
    "!osm_tags": {
       "access": ["no","private"]
    },
    "osm_element": [
       "OpenWay",
       "ClosedWay"
    ]
 }
```

Or we could alter the conditions for the "full" surface question like this:

```json
 {
    "osm_tags": {
       "highway": "footway",
       "footway": [false, "sidewalk"],
       "surface": false,
       "indoor": ["no", false],
       "footway:surface": false,
       "cycleway:surface": false,
       "segregated": ["no", false]
    },
    "!osm_tags": {
       "access": ["no","private"]
    },
    "osm_element": [
       "OpenWay",
       "ClosedWay"
    ]
},
```

Basically this would not ask the questions when a sub surface tag is set (makes it more strict) but still ask the question when segregated is unknown (making it less strict).

File: assets/datasets/question_catalog.json [hunk 8]
```diff
@@ -1740,6 +1740,50 @@
...
+         ]
+      },
+      "conditions": [
+         {
+            "osm_tags": {
+               "highway": ["path", "footway", "cycleway"],
+               "segregated": false,
+               "foot": "designated",
+               "bicycle": "designated",
+               "indoor": ["no", false],
```
Comment (Robbendebiene) [inline assets/datasets/question_catalog.json:1775]: Btw. this comma is causing the validation to fail. While this is valid in JavaScript, for strict JSON validation trailing commas are considered invalid.

**Comment by Robbendebiene** [comment]
> Wouldn't this disable both questions on a way with e.g. highway=path + foot=designated + bicycle=designated + segregated=yes? Both questions would depend on the other being answered already, so I think this doesn't work.

I think there is a miss-understanding here. I don't want to replace or alter the current condition instead I want to add another one (like we have multiple for the surface question). We use the term "condition" for every direct item denoted with `{}` in the `conditions` property which can be confusing sometimes.

So for the footway question it would look like this:

```json
  {
     "question": {
         "name": "Bodenbeschaffenheit Fußweg",
         "text": "Wie ist der Boden des Fußweges beschaffen?",
         "description": "Besteht die Oberfläche aus Schotter, Sand, Gras oder einem vergleichbaren Belag, so handelt es sich um einen unbefestigten Bodenbelag."
      },
      "answer": {
         "type": "List",
         "input": [
            {
               "name": "Befestigt",
               "image": "assets/images/questions/surface_paved.jpg",
               "osm_tags": {
                  "footway:surface": "paved"
               }
            },
            {
               "name": "Unbefestigt",
               "image": "assets/images/questions/surface_unpaved.jpg",
               "osm_tags": {
                  "footway:surface": "unpaved"
               }
            }
         ]
      },
      "conditions": [
         {
            "osm_tags": {
               "highway": ["path", "footway", "cycleway"],
               "segregated": "yes",
               "surface": false,
               "footway:surface": false,
               "indoor": ["no", false]
            },
            "!osm_tags": {
               "access": ["no","private"]
            },
            "osm_element": [
               "OpenWay",
               "ClosedWay"
            ]
         },
        {
            "osm_tags": {
               "highway": ["path", "footway", "cycleway"],
               "cycleway:surface": true,
               "footway:surface": false,
               "indoor": ["no", false]
            },
            "!osm_tags": {
               "access": ["no","private"]
            },
            "osm_element": [
               "OpenWay",
               "ClosedWay"
            ]
         }
      ]
 }
```


For the cycleway question it would look like this:

```json
  {
      "question": {
         "name": "Bodenbeschaffenheit Radweg",
         "text": "Wie ist der Boden des Radweges beschaffen?",
         "description": "Besteht die Oberfläche aus Schotter, Sand, Gras oder einem vergleichbaren Belag, so handelt es sich um einen unbefestigten Bodenbelag."
      },
      "answer": {
         "type": "List",
         "input": [
            {
               "name": "Befestigt",
               "image": "assets/images/questions/surface_paved.jpg",
               "osm_tags": {
                  "cycleway:surface": "paved"
               }
            },
            {
               "name": "Unbefestigt",
               "image": "assets/images/questions/surface_unpaved.jpg",
               "osm_tags": {
                  "cycleway:surface": "unpaved"
               }
            }
         ]
      },
      "conditions": [
         {
            "osm_tags": {
               "highway": ["path", "footway", "cycleway"],
               "segregated": "yes",
               "surface": false,
               "cycleway:surface": false,
               "indoor": ["no", false]
            },
            "!osm_tags": {
               "access": ["no","private"]
            },
            "osm_element": [
               "OpenWay",
               "ClosedWay"
            ]
         },
        {
            "osm_tags": {
               "highway": ["path", "footway", "cycleway"],
               "footway:surface": true,
               "cycleway:surface": false,
               "indoor": ["no", false]
            },
            "!osm_tags": {
               "access": ["no","private"]
            },
            "osm_element": [
               "OpenWay",
               "ClosedWay"
            ]
         }
      ]
},
```

File: assets/datasets/question_catalog.json [hunk 9]
```diff
@@ -1797,6 +1844,97 @@
...
+               "image": "assets/images/questions/surface_paved.jpg",
+               "osm_tags": {
+                  "footway:surface": "paved"
+               }
+            },
+            {
+               "name": "Unbefestigt",
+               "image": "assets/images/questions/surface_unpaved.jpg",
+               "osm_tags": {
+                  "footway:surface": "unpaved"
```
[the remark this answers is not in the document]
Comment (Henry00572) [reply assets/datasets/question_catalog.json]: Oh, I changed that now.

**Comment by Henry00572** [comment]
> I want to add another one (like we have multiple for the surface question). We use the term "condition" for every direct item denoted with {} in the conditions property which can be confusing sometimes.

Thanks for clearing that up, I understand now and have added this condition.

**Review by Robbendebiene — APPROVED** [review]
LGTM!

Thank you for your contribution 🙂

**Comment by Henry00572** [comment]
Thanks for all the help!
